lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -815,7 +818,17 @@ def _start_opensearch(self, event: _StartOpenSearch) -> None:  # noqa: C901
         if self.opensearch.is_started():
             try:
                 self._post_start_init(event)
-            except (OpenSearchHttpError, OpenSearchNotFullyReadyError):
+            except (
+                OpenSearchHttpError,
+                OpenSearchStartTimeoutError,
+                OpenSearchNotFullyReadyError,
```

Comment:
`_post_start_init` doesn't raise a `OpenSearchStartTimeoutError`
```suggestion
                OpenSearchHttpError,
                OpenSearchNotFullyReadyError,
```